share/spice/movies/cast/movies_cast.js

```diff
@@ -0,0 +1,70 @@
+(function (env) {
+    'use strict';
+
+    // This is needed to prevent jQuery from adding '_={timestamp}' to the end
+    // of the URL, which breaks the call to getScript()
+    $.ajaxSetup({ cache: true });
+
+    var DETAIL_ENDPOINT = '/js/spice/movies/details/';
+    var POSTER_URL = 'http://image.tmdb.org/t/p/w185';
+    var MOVIE_URL = 'http://www.themoviedb.org/movie/';
+
+    env.ddg_spice_movies_cast = function(data) {
+        // This isn't exactly pretty, but I can't think of any
+        // more graceful way to handle this.
+        var hasFailed = (
+            !data ||
+            !data.results ||
+            !data.results[0] ||
+            !data.results[0].id
+        );
+
+        if (hasFailed) {
+            return Spice.failed('movies_cast');
+        }
+
+        $.getJSON(DETAIL_ENDPOINT + data.results[0].id, function(api_result) {
+            var hasFailed = (
+                !api_result ||
+                !api_result.results ||
+                !api_result.results.length
+            );
+
+            if (hasFailed) {
+                return Spice.failed('movies_cast');
+            }
+
+            Spice.add({
+                id: 'cast',
+                name: 'Movies',
+                data: api_result.results,
+                meta: {
+                    sourceName: 'TheMovieDb',
+                    sourceUrl: 'http://www.themoviedb.org/person/'
+                        + api_result.results[0].id,
```

Comment:
Got it. Thanks guys, and I apologize for the delay. I'll send out a patch
for these issues this week, if that's okay.

On Mon, Jun 13, 2016 at 12:05 PM, Travis Bell <notifications@github.com>
wrote:

> In share/spice/movies/cast/movies_cast.js
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2762#discussion_r66848458>
> :
>
> > +                !api_result.results ||
> > +                !api_result.results.length
> > +            );
> > +
> > +            if (hasFailed) {
> > +                return Spice.failed('movies_cast');
> > +            }
> > +
> > +            Spice.add({
> > +                id: 'cast',
> > +                name: 'Movies',
> > +                data: api_result.results,
> > +                meta: {
> > +                    sourceName: 'TheMovieDb',
> > +                    sourceUrl: 'http://www.themoviedb.org/person/'
> > +                        + api_result.results[0].id,
>
> Aha! I managed to click the second "Movies" IA and got the invalid URL I
> was talking about:
> http://files.travisbell.com/movies_starring_johnny_depp_at_DuckDuckGo_and_Inbox_and_tv_controller_rb_—_www_themoviedb_org_1D0F3B3D.png
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-spice/pull/2762/files/5d990aa0544d164a82fb14b184481b42055055b0#r66848458>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAbryzRCz_8-cEUVTuIWU1lmrOhvfbO6ks5qLanugaJpZM4Iovxx>
> .
>